ui/hooks/useAssetDetails.js

```diff
@@ -38,6 +39,21 @@ export function useAssetDetails(tokenAddress, userAddress, transactionData) {
         transactionData,
         nfts,
       );
+      const token = util.getContractAtAddress(tokenAddress);
+
+      const symbol = await token.symbol();
+      const name = await token.name();
```

Comment:
I'm assuming these are RPC requests, in which case `Promise.all` can be used to issue them in parallel instead of sequentially.